### PR TITLE
post legacy layout

### PR DESCRIPTION
**Pull request by HamdanSCode**
some todos in there because there is a pr up that adds some strings it needs. Will also readjust some of the margins and padding after the uiupdate is merged in.

### CODE REVIEW

File: applegacy/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -26,4 +26,17 @@
         </activity>
         <activity android:name="com.hamdan.forzenbook.legacy.createaccount.LegacyCreateAccountActivity" />
     </application>
-</manifest>
\ No newline at end of file
+</manifest>
+
+<!--<activity-->
```
Comment (ForzenAcademy) [inline applegacy/src/main/AndroidManifest.xml]: Why leave here?

File: appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt [hunk 2]
```diff
@@ -10,4 +10,7 @@ class NavigatorImpl : Navigator {
 
     override fun navigateToCreateAccount(context: Context) {
     }
+
+    override fun navigateToPost(context: Context) {
```
Comment (ForzenAcademy) [inline appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt:14]: TODOs for these? Why empty?

File: postCompose/src/main/java/com/hamdan/forzenbook/post/compose/Post.kt [hunk 3]
```diff
@@ -209,9 +209,9 @@ private fun StandardContent(
                 horizontalArrangement = Arrangement.Center,
             ) {
                 PillToggleSwitch(
-                    imageLeftRes = R.drawable.baseline_text_fields_24,
+                    imageLeftRes = R.drawable.pill_text_white,
                     leftDescriptionRes = R.string.text_toggle_text,
-                    imageRightRes = R.drawable.image_post_icon,
+                    imageRightRes = R.drawable.pill_image_white,
```
Comment (ForzenAcademy) [inline postCompose/src/main/java/com/hamdan/forzenbook/post/compose/Post.kt]: `white`? What about dark mode?

File: postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt [hunk 4]
```diff
@@ -14,9 +14,9 @@ class PostRepositoryImpl(
     override suspend fun postText(token: String, message: String) {
         val response = service.sendTextPost(token, TEXT_TYPE, message)
         if (!response.isSuccessful) {
-            if(response.code() == 401){
-                throw(NetworkRetrievalException())
-            }else{
+            if (response.code() == 401) {
```
Comment (ForzenAcademy) [inline postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt]: Constant

File: postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt [hunk 5]
```diff
@@ -33,9 +33,9 @@ class PostRepositoryImpl(
             MultipartBody.Part.createFormData("fileToUpload", file.name, requestBodyFile)
         )
         if (!response.isSuccessful) {
-            if(response.code() == 401){
-                throw(NetworkRetrievalException())
-            }else{
+            if (response.code() == 401) {
```
Comment (ForzenAcademy) [inline postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt]: constant

File: postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt [hunk 6]
```diff
@@ -33,9 +33,9 @@ class PostRepositoryImpl(
...
+                throw (NetworkRetrievalException())
```
Comment (ForzenAcademy) [inline postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt]: These 401s are unauthorized exceptions. This error sounds generic

File: uiCore/src/main/res/drawable/pill_image_selector.xml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item android:drawable="@drawable/pill_image_white" android:state_checked="true" />
```
Comment (ForzenAcademy) [inline uiCore/src/main/res/drawable/pill_image_selector.xml]: Make drawable attributes so this scales for theme?

File: applegacy/src/main/AndroidManifest.xml (hunk 1 above)
Comment (HamdanSCode) [reply applegacy/src/main/AndroidManifest.xml]: yea its not needed, I should've put a todo for when the legacy post is finished to put it back, but honostly its not needed to easily put it back.

File: appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt (hunk 2 above)
Comment (HamdanSCode) [reply appLegacyFragment/src/main/java/com/hamdan/forzenbook/legacy/fragment/app/view/NavigatorImpl.kt:14]: theres a comment above the class:
"Need an empty impl of Navigator for dagger to be happy"
Should we try to seperate where the navigators for fragment and legacy are created then?
Currently both fragment navigator and Navigator are interfaces in legacyCore that are both expected in any legacy based app. Ideally only the navigator goes to legacy apps that are fragments and the fragment navigator goes to fragment apps.
I could split legacyCore into two modules then potentialy of legacyCore and fragmentLegacyCore

File: postCompose/src/main/java/com/hamdan/forzenbook/post/compose/Post.kt (hunk 3 above)
Comment (HamdanSCode) [reply postCompose/src/main/java/com/hamdan/forzenbook/post/compose/Post.kt]: updated, wasn't sure if the reference fix would worked but thankfuly it did, changed the tinting of the images to be based on the color attributes. it is now selected and unselected.

File: postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt (hunk 6 above)
Comment (HamdanSCode) [reply postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt]: I think your right and I also think I can throw a different error of InvalidTokenException as that should be the only thing that happens in that case

File: uiCore/src/main/res/drawable/pill_image_selector.xml (hunk 7 above)
Comment (HamdanSCode) [reply uiCore/src/main/res/drawable/pill_image_selector.xml]: are you referring to something like this:
https://stackoverflow.com/questions/3487247/format-attribute-value-androiddrawable-not-valid ?
I'm not seeing a drawable format so I'm assuming I'd have to do it in some way through either reference or something else.

File: postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt (hunk 6 above)
Comment (ForzenAcademy) [reply postCore/src/main/java/com/hamdan/forzenbook/post/core/data/repository/PostRepositoryImpl.kt]: *you're

**Review by ForzenAcademy — APPROVED** [review]

**Pull request merged**